cf-check/cf-check.c

```diff
@@ -53,12 +101,71 @@ int main(int argc, const char *const *const argv)
     {
         print_help();
         Log(LOG_LEVEL_ERR, "No command given");
-        return 1;
+        return EXIT_FAILURE;
     }
 
-    const int cmd_argc = argc - 1;
-    const char *const *const cmd_argv = argv + 1;
-    const char *const command = cmd_argv[0];
+    int c = 0;
+    int start_index = 1;
+    const char *optstr = "+hMg:dvI"; // + means stop for non opt arg. :)
+    while ((c = getopt_long(argc, (char *const *) argv, optstr, OPTIONS, &start_index))
+            != -1)
+    {
+        switch (c)
+        {
+            case 'd':
+            {
+                LogSetGlobalLevel(LOG_LEVEL_DEBUG);
+                break;
+            }
+            case 'v':
+            {
+                LogSetGlobalLevel(LOG_LEVEL_VERBOSE);
+                break;
+            }
+            case 'I':
+            {
+                LogSetGlobalLevel(LOG_LEVEL_INFO);
+                break;
+            }
+            case 'g':
+            {
+                LogSetGlobalLevelArgOrExit(optarg);
+                break;
+            }
+            case 'V':
+            {
+                print_version();
+                return EXIT_SUCCESS;
+                break;
+            }
+            case 'h':
+            {
+                print_help();
+                return EXIT_SUCCESS;
+                break;
+            }
+            case 'M':
+            {
+                Writer *out = FileWriter(stdout);
+                ManPageWrite(out, "cf-check", time(NULL),
+                             CF_CHECK_SHORT_DESCRIPTION,
+                             CF_CHECK_MANPAGE_LONG_DESCRIPTION,
+                             OPTIONS, HINTS,
+                             true);
+                FileWriterDetach(out);
+                return EXIT_SUCCESS;
+                break;
+            }
+            default:
+            {
+                return EXIT_FAILURE;
+                break;
+            }
+        }
+    }
+    const char *const *const cmd_argv = &(argv[optind]);
```

Comment:
I think this is worth adding as a comment:
> When ‘getopt_long’ has no more options to handle, it returns ‘-1’,
> and leaves in the variable ‘optind’ the index in ARGV of the next
> remaining argument.

Reply:
Also, I think `argv + optind` would be better readable than the `&(a[i])` expression.

Reply:
Alright.